build-in-container/scripts/setup.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+echo "------------ Setting up Mariner Build Environment ------------"
+
+# build variables
+MARINER_BASE_DIR=/mariner
+BUILD_DIR="$MARINER_BASE_DIR/build"
+CCACHE_DIR="$MARINER_BASE_DIR/ccache"
+CBL_MARINER_GIT_URL="https://github.com/microsoft/CBL-Mariner.git"
+CHROOT_DIR="/temp/DockerStage/"
+DISABLE_UPSTREAM_REPOS="n"
+IMAGE_CONFIG_DIR="$MARINER_BASE_DIR/toolkit/imageconfigs"
+LOG_LEVEL="info"
+LOG_PUBLISH_DIR="$MARINER_BASE_DIR/logs"
+MARINER_RELEASE_TAG="2.0-stable"
+OUT_DIR="$MARINER_BASE_DIR/out"
+RUN_CHECK="y"
+SOURCE_URL="https://cblmarinerstorage.blob.core.windows.net/sources/core"
+SPECS_DIR="$MARINER_BASE_DIR/SPECS"
+USE_CCACHE="y"
+
+# Build Mariner toolkit if not present, by cloning Mariner GitHub repo
+#
+# No arguments
+# Global variables expected to be defined: BUILD_DIR, CHROOT_DIR, CHROOT_NB, OUT_DIR
+download_mariner_toolkit() {
+    if [ ! "$(ls -A toolkit)" ]; then
+    echo "------------ Preparing Mariner toolkit ------------"
+        if [ ! -d CBL-Mariner ]; then
```

Comment:
We should also check if the tag is correct or not. If this environment is being resused `2.0-stable` will become out of date. Need to run `git fetch --tags` to get updates, then check if the commit is still correct. If it isn't we should regenerate the toolkit.

Reply:
We're not promising that the specs will be built using the latest 2.0-stable toolkit. I think Reuben had pointed out caching of toolkit might be injurious, but I am planning to take that up in the next revision of this tool

Reply:
ok, its fine to wait on this. We can suggest using the tidy if a new toolkit is needed.